src/Analysis/Ast/Impl/Analyzer/Handlers/SequenceExpressionHandler.cs

```diff
@@ -65,20 +65,29 @@ internal sealed class SequenceExpressionHandler : StatementHandler {
                     case NameExpression nex when !string.IsNullOrEmpty(nex.Name):
                         eval.DeclareVariable(nex.Name, valueEnum.Next, VariableSource.Declaration, nex);
                         break;
+                    // Nested sequence expression in sequence, Tuple[Tuple[int, str], int], List[Tuple[int], str]
+                    // TODO: Because of bug with how collection types are constructed, they don't make nested collection types
+                    // into instances, meaning we have to create it here
+                    case SequenceExpression se when valueEnum.Peek is IPythonCollection || valueEnum.Peek is IPythonCollectionType:
```

Comment:
You may be able to `case SequenceExpression se when valueEnum.Next is IPythonCollection pc || valueEnum.Next is IPythonCollectionType pct:`

Reply:
I don't think I can - calling next will shift the index up one so the next time we call next it will have a different value